sims/networks/basic/timing/GunnsRosesTiming.cpp

```diff
@@ -115,49 +124,58 @@ void GunnsRosesTiming::update(const double timestep)
                     N += 2;
                 }
             } else if (Gunns::GPU_DENSE == gpuMode) {
+                // Handle GPU DENSE mode
                 gpuTimePrev = gpuTime;
                 gpuTime = network->mSolveTimeAvg;
                 printf("    %9.2e", network->mSolveTimeAvg);
+                
                 if (errCheck) {
                     double error = 0.0;
                     for (unsigned int i=0; i<5; ++i) {
                         error += std::fabs(gpuPotentials[i] - cpuPotentials[i]);
                     }
                     printf("  %9.2e", error);
                 }
-                step = 0;
                 predictGpuThreshold();
                 gpuMode = Gunns::GPU_SPARSE;
             } else if (Gunns::GPU_SPARSE == gpuMode) {
+                // Handle GPU SPARSE mode
                 gpuSparseTimePrev = gpuSparseTime;
                 gpuSparseTime = network->mSolveTimeAvg;
                 printf("    %9.2e", network->mSolveTimeAvg);
+                
                 if (errCheck) {
                     double error = 0.0;
-                    for (unsigned int i=0; i<5; ++i) {
-                        error += std::fabs(gpuSparsePotentials[i] - cpuPotentials[i]);
+                    for (unsigned int i = 0; i < 5; ++i) {
+                        error += fabs(gpuSparsePotentials[i] - cpuPotentials[i]);
                     }
                     printf("  %9.2e", error);
                 }
-                printf("\n");
-                step = 0;
                 predictGpuSparseThreshold();
                 gpuMode = Gunns::NO_GPU;
                 N += 2;
             }
+            
+            step = 0;
         }
     } else {
         printf("----------------------------------------------\n");
         if (gpuEnabled) {
             printf("GPU # nodes\n");
             printf("threshold estimates:      %7.0f      %7.0f\n", gpuThreshold, gpuSparseThreshold);
-            printf("\n");
-            printf("Your network's times and\n");
-            printf("best thresholds may vary.\n");
+            printf("\nYour network's times and best thresholds may vary.\n");
             printf("----------------------------------------------\n");
         }
+        
         printf("\nTerminating sim:\n");
         exec_terminate("", "");  // from Trick's exec_proto.h, tell Trick to kill the sim
+
```

Comment:
Please delete lines 172-178 and resubmit.  Execution never reaches this point because once exec_terminate is called above, Trick terminates the sim executable and execution never returns to this caller.  Also, there is no reason to divide N or any other variable by step, so there is no reason to check for step = 0 here anyway.  You said "N /= step;: If step is non-zero, it divides N by step. This may be a part of some final computation or cleanup process related to the number of steps or nodes (N)."  However, there is no final computation or cleanup needed -- the sim execution is already ended at this point.

Reply:
Removed the unnecessary division of N by step
Removed the division-by-zero check
Kept the clean program termination with exec_terminate()